packages/flutter/test/widgets/buttons_test.dart

```diff
@@ -40,4 +41,102 @@ void main() {
 
     semantics.dispose();
   });
+
+
+  Widget _buttonWidget({Key buttonKey, Key materialKey,
+    Color color, Color highlightColor, Color splashColor,
+    double minWidth = 150.0, double height = 60.0, bool useTheme = false}) {
+
+    final Key definedMaterialKey = materialKey ?? new UniqueKey();
+    final Key definedButtonKey = buttonKey ?? new UniqueKey();
+
+    Widget buttonWidget = new Material(
+      key: definedMaterialKey,
+      child: new Center(
+        child: new MaterialButton(
+          key: definedButtonKey,
+          color: color,
+          highlightColor: !useTheme ? highlightColor : null,
+          splashColor: !useTheme ? splashColor : null,
+          minWidth: minWidth,
+          height: height,
+          onPressed: () { },
+        ),
+      ),
+    );
+    if (useTheme) {
+      final ThemeData themeData = new ThemeData(
+        accentColor: color,
+        highlightColor: highlightColor,
+        splashColor: splashColor,
+      );
+      buttonWidget = new Theme(
+        data: themeData,
+        child: buttonWidget,
+      );
+    }
+    return buttonWidget;
+  }
+
+  testWidgets('Does button highlight + splash colors work if set directly', (WidgetTester tester) async {
+    final Color buttonColor = new Color(0xFFFFFF00);
+    final Color highlightColor = new Color(0xDD0000FF);
+    final Color splashColor = new Color(0xAA0000FF);
+
+    final Key materialKey = new UniqueKey();
+    final Key buttonKey = new UniqueKey();
+
+    await tester.pumpWidget(
+      _buttonWidget(
+        materialKey: materialKey,
+        buttonKey: buttonKey,
+        color: buttonColor,
+        highlightColor: highlightColor,
+        splashColor: splashColor,
+      ),
+    );
+
+    await tester.longPress(find.byKey(buttonKey));
+    await tester.pump(new Duration(milliseconds: 300));
+
+    final RRect buttonRRect = new RRect.fromLTRBR(0.0, 0.0, 150.0, 60.0, new Radius.circular(2.0));
+    // TODO(lukef) - the object returned by renderObject does not contain splash or highlights (??)
+    expect(tester.renderObject(find.byKey(buttonKey)), paints
```

Comment:
I would expect the object that contains the splash and highlight to be the Material whose child is the button. The button doesn't paint its own splashes and highlights if it has no elevation, they get painted by the parent Material.